backend/app/anti_recommenders/arkg/arkg_anti_recommender.py

```diff
@@ -1,162 +1,209 @@
-import typing
+import logging
 from collections.abc import Iterable
 from pathlib import Path
+from typing import override
 
 import pyoxigraph as ox
+from postgrest import APIError
 
 from app.anti_recommenders import AntiRecommender
-from app.constants import WIKIPEDIA_BASE_URL
-from app.models import AntiRecommendation, AntiRecommendationGraph
+from app.constants import (
+    ARKG_ANTI_RECOMMENDER_USER_STATE_TABLE_NAME,
+    WIKIPEDIA_BASE_URL,
+)
+from app.database import database_client
+from app.models import AntiRecommendation, User, UserState
 from app.models.types import RdfMimeType, RecordKey, StoreQuery
 from app.namespaces import SCHEMA
 
 
 class ArkgAntiRecommender(AntiRecommender):
     """
-    A concrete implementation of AntiRecommender.
+    A concrete implementation of `AntiRecommender`.
 
-    ArkgAntiRecommender uses information stored in an Anti-Recommendation Knowledge Graph to generate AntiRecommendations.
+    `ArkgAntiRecommender` uses information stored in an Anti-Recommendation Knowledge Graph to generate `AntiRecommendations`.
     """
 
     def __init__(
         self,
         *,
-        arkg_base_iri: ox.NamedNode,
-        arkg_file_path: Path,
-        arkg_mime_type: RdfMimeType,
+        base_iri: ox.NamedNode,
+        file_path: Path,
+        mime_type: RdfMimeType,
         record_keys: tuple[RecordKey, ...],
+        user: User,
     ) -> None:
-        self.__anti_recommendation_path_graph: list[AntiRecommendationGraph] = []
-        self.__arkg_base_iri = arkg_base_iri
-        self.__arkg_file_path = arkg_file_path
-        self.__arkg_mime_type = arkg_mime_type
+        self.__base_iri = base_iri
+        self.__logger = logging.getLogger(__name__)
         self.__record_keys: tuple[RecordKey, ...] = record_keys
-        self.__store: ox.Store = ox.Store()
-        self.__load_store()
+        self.__store: ox.Store = self.__load_store(
+            file_path=file_path, mime_type=mime_type, base_iri=base_iri
+        )
+        self.__user_state: UserState | None = self.__fetch_user_state_from_database(
+            user
+        )
 
-    def __create_query(self, record_key: RecordKey) -> StoreQuery:
+    def __fetch_anti_recommendations_query(self, record_key: RecordKey) -> StoreQuery:
```

Comment:
Feels weird to have `fetch_` be a verb in the method name below and part of a noun phrase here. I'd probably just inline this method to where it's used rather than have this awkward naming. The method only seems to be called in one place.